### DISCUSSION TITLE
posts/2022/07/really-getting-started-hugo/

### DISCUSSION THREAD

**Discussion by giscus** [brycewray/comments#21] [Announcements]
# posts/2022/07/really-getting-started-hugo/

Since the officially recommended approach tends to frustrate new users, let’s see whether we can do something about that.

https://www.brycewray.com/posts/2022/07/really-getting-started-hugo/

**Comment by hiAndrewQuinn** [comment]
This is an excellent post, and I appreciate the callout. ( I had good reason to stick the submodule part only at the end ;) )

The theme-less approach you lay out here is fresh to my eyes; it makes sense to me now that I've seen where you put what, but I honestly always thought you had to have _some_ kind of theme attached to a website, no matter how barren. It has still been a pain point for me any time I come back to messing with Hugo after a long time away that `index.html` or `_default/single.html` don't come with a `{{ .Content }}` tag by default, and I'm glad you put those right front and center in the texty part of your tutorial.

I read your [About](https://www.brycewray.com/about/) and felt a kindred spirit there, too. I wasn't around for the days of Geocities and Myspace, and indeed part of the reason I took to SSGs in particular was because they seemed to scratch that itch of having a tiny virtual home of your own where you could just sling up some HTML and have fun.

**Reply by brycewray** [reply]
@hiAndrewQuinn Thanks! If what I’ve termed the “themes-first/defaults-deficient approach” continues to be the Hugo norm, one like yours certainly should be high in the list of those from which new users should choose.

**Comment by bernarddt** [comment]
Oh there is no way! I've been sitting studying Hugo since Yesterday morning (no I did sleep, but the point is I started Yesterday morning and spent the day on this). 

After I've read probably 50% of the docs, went through the example, getting a quick page up and running, watched 2 x video tutorials recommended by the docs (strangely they seem to be outdated a bit, as they talk about _index.html or _index.md files, not sure which it was, that seems to have changed.)

Anyways, part of my research Yesterday was googling for tutorials, ok to be exact I googled: "install and use hugo for visual studio 2022", but got no good results, but this article's title made me think it would be a good one to read down the line when I'm ready to actually start with a "real" website, not just building quick "blog" example sites.

Long story short, after 8 hours of reading and watching, I started last night trying to build the website. I have a non-Hugo template/theme that I want to use, and thought... I would just convert that to a Hugo theme, then realised, that maybe I should not do that, but rather just build a site "WITHOUT A THEME". That was probably madness kicking in, since every article, even ChatGPT (ok ok, it was Github Copilot) recommended that.

But funny enough I could not get the site to build a simple page with no templating in it, and without going back and reading a lot of literature again, I thought I'll pick up this morning....

Then this morning I remembered this page I cam across and put aside for when I'm "ready". And low and behold... it addresses the exact problem I'm having! It is now 2024, and Hugo have not made any changes to the "use a theme" proposal.

The funny thing is, Hugo explain that a theme is simply a "insert-fancy-word" yaml container like merging of the themes folder with you project (ok it is as if the theme files are just copied into your project folder. So technically you "don't" need a theme, your site can be the theme itself. 

This is getting long, I want to point out that your article is exactly what I need, I haven't even read the whole thing, Yesterday it did not stand out much more than looking like a more "real world" example. And this morning just reading the TLDR part, I can see it addresses exactly what I was looking for.

Just a side note: My frustration with Hugo tutorials and advice, is that the Hugo system seems to be like WordPress 10 years ago. It was focussed primarily on "Blog" sites, where you have a template and then just pages and pages of content. And most if not all (I've yet to find one) of the YouTube tutorials are based on how to start a Blog site. Most YouTubers seem to come from a background where they do have a blog site that they want to convert to Hugo. 

So effectively there are very little to no tutorials on Hugo that use it to generate a business website. And just like WordPress had to take a long journey to get from a "Blog" site CMS to a "Web" site CMS, Hugo will probably also take 10 years to adjust. 

I for instance wanted to have the feature of "blocks" in WordPress that are now first class citizens so you can easily pick a block style and then fill the content. With Hugo it will seem that ability don't exist yet, but I'm still learning. Trying to do something against the current normally means you have to work extra hard to learn the system and then use it for what it was not meant 😊

Cheers!

**Comment by bernarddt** [comment]
Sorry for the long previous post, but I just have to add, I'm now reading the article, up to just before "Step1". 

Basically where you mention that a newbie have to get to that "aha" moment. The problem with the theme first approach is the "black box" problem. Depending on your goal, you either like the black box approach or it will completely hinder the leaning experience.

I'm learning Hugo as a developer/designer/website builder. Not as a user. The difference is, I KNOW html and css. That is the fundamental building blocks of a website. Users on the other hand, KNOW WordPress or some other CMS (black box). They don't care how it work, they just want to edit and add content. The designer/builder/developer setup up the black box and have to resolve issues when the black box fails (WordPress). 

So I do need to understand "step-by-step" how you get from a HTML + CSS page (or maybe two) to the Hugo ecosystem (Hugo's black box).  And by jumping in with a "theme" breaks that process. I would be much more efficient if it would start with a simple HTML + CSS page (or two), and then gradually break it up into each component, moving each part t the relevant location/template/layout area.

This way we can see how each part or feature of Hugo "improves" or simplify the structure of the site and how each part is broken up.

It is not a new approach, we use it daily while coding large systems. You first code a "proof-of-concept" function, basically a large obnoxious function. Then you iteratively start to break it up into functions, simplifying and improving the code (ok I must admit that 50% of the coders out there would stop after the first step with a "but it works" excuse).  

It is important to know.... we don't like magic. Magic is the thing that will keep you up all night trying to resolve an issue why the magic stopped working after 2 years on a Friday night just after everyone went home.
